src/firebolt/common/base_cursor.py

```diff
@@ -367,7 +368,11 @@ def fetchall(self) -> List[List[ColType]]:
         left, right = self._get_next_range(self.rowcount)
         assert self._rows is not None
         rows = self._rows[left:right]
-        return [self._parse_row(row) for row in rows]
+
+        with Timer("[PERFORMANCE] Parsing query output into native Python types "):
+            result = [self._parse_row(row) for row in rows]
+
+        return result
```

Comment:
We would want this in `fetchone` and `fetchmany` functions as well since users could be fetching results in chunks.